fix(swap)!: build paraswap tx error

saledjenic opened the pull request:
The first commit is about retrying logic for the build tx call. If an error occurs, the app will try to make the same call up to three times, while the last one will update the fetched prices first and then make the build tx call.

The second commit is solving a real issue, cause the main reason, based on the response we got from the paraswap team, for the build tx error that was happening not always but more or less often, was missed slippage param. 

In the past slippage parameter was provided at a later stage, when the tx was about to be built for real, but now, since we fetch the tx data from the paraswap API at the moment we do the gas estimation, the slippage parameter is necessary earlier. That's why we have to provide it at the moment of fetching the route and also clients need to re-fetch the route on every slippage change.

What's changed:
- Removed the `RouterBuildTransactionsParams` type and its usage across the codebase.
- Updated `BuildTransactionsFromRoute` function signatures to use `uuid` instead of the removed type.
- Adjusted related database operations and tests to reflect the removal of build input parameters.
- `SwapParaswapProcessor` updated to use slippage param appropriately, reduced calls to paraswap api, by storing tx data


What you need to do on the client side is:
- to update `wallet_buildTransactionsFromRoute` call, cause it needs only `uuid` now as single param (no object of uuid and slippage anymore)
- to provide `slippagePercentage` as float when doing `GetSuggestedRoutesAsync` call
- to do `GetSuggestedRoutesAsync` call whenever the user changes the slippage (the same as we do when the user changes the amount or selected token)

saledjenic committed:
fix(swap)_: build paraswap tx with retry instead of just build

If an error occurs, usually it's `Unable to build transaction (1001)` error the logic will try to do the same
call up to 3 times, while the last one will update the fetched prices as well.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (48)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 0e7c5b9a | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6548/1/) | 2025-04-22 21:20:19 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250422-211739-0e7c5b-pr6548.zip) |
| :heavy_check_mark: | 0e7c5b9a | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6548/1/) | 2025-04-22 21:20:42 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250422-211738-0e7c5b-pr6548.aar) |
| :heavy_check_mark: | 0e7c5b9a | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6548/1/) | 2025-04-22 21:21:41 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250422-211742-0e7c5b-pr6548.zip) |
| :heavy_check_mark: | 0e7c5b9a | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6548/1/) | 2025-04-22 21:22:24 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250422-211747-0e7c5b-pr6548.zip) |
| :heavy_check_mark: | 0e7c5b9a | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6548/1/) | 2025-04-22 21:23:21 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250422-211744-0e7c5b-pr6548.zip) |
| :heavy_check_mark: | 0e7c5b9a | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6548/1/) | 2025-04-22 21:23:33 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250422-211741-0e7c5b-pr6548.zip) |
| :heavy_check_mark: | 0e7c5b9a | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/1/) | 2025-04-22 21:27:53 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/1//consoleText) |
| :heavy_check_mark: | 0e7c5b9a | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/1/) | 2025-04-22 21:54:37 | ~36 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e331d471 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6548/2/) | 2025-04-24 13:03:05 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250424-130025-e331d4-pr6548.zip) |
| :heavy_check_mark: | e331d471 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6548/2/) | 2025-04-24 13:03:23 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250424-130022-e331d4-pr6548.aar) |
| :heavy_check_mark: | e331d471 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6548/2/) | 2025-04-24 13:03:44 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250424-130034-e331d4-pr6548.zip) |
| :heavy_check_mark: | e331d471 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6548/2/) | 2025-04-24 13:04:28 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250424-130027-e331d4-pr6548.zip) |
| :heavy_check_mark: | e331d471 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6548/2/) | 2025-04-24 13:06:11 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250424-130030-e331d4-pr6548.zip) |
| :heavy_check_mark: | e331d471 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6548/2/) | 2025-04-24 13:06:17 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250424-130025-e331d4-pr6548.zip) |
| :heavy_multiplication_x: | e331d471 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/2/) | 2025-04-24 13:10:43 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/2//consoleText) |
| :heavy_check_mark: | e331d471 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/2/) | 2025-04-24 13:35:36 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | a5c4d583 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6548/3/) | 2025-04-24 13:09:44 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250424-130707-a5c4d5-pr6548.zip) |
| :heavy_check_mark: | a5c4d583 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6548/3/) | 2025-04-24 13:10:07 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250424-130705-a5c4d5-pr6548.aar) |
| :heavy_check_mark: | a5c4d583 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6548/3/) | 2025-04-24 13:10:17 | ~2 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250424-130717-a5c4d5-pr6548.zip) |
| :heavy_check_mark: | a5c4d583 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6548/3/) | 2025-04-24 13:11:08 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250424-130710-a5c4d5-pr6548.zip) |
| :heavy_check_mark: | a5c4d583 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6548/3/) | 2025-04-24 13:12:53 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250424-130713-a5c4d5-pr6548.zip) |
| :heavy_check_mark: | a5c4d583 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6548/3/) | 2025-04-24 13:13:03 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250424-130708-a5c4d5-pr6548.zip) |
| :heavy_multiplication_x: | a5c4d583 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/3/) | 2025-04-24 13:20:53 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/3//consoleText) |
| :heavy_check_mark: | a5c4d583 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/3/) | 2025-04-24 14:09:39 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c683e0c3 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6548/4/) | 2025-04-25 10:34:26 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250425-103144-c683e0-pr6548.zip) |
| :heavy_check_mark: | c683e0c3 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6548/4/) | 2025-04-25 10:34:44 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250425-103143-c683e0-pr6548.aar) |
| :heavy_check_mark: | c683e0c3 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6548/4/) | 2025-04-25 10:35:15 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250425-103152-c683e0-pr6548.zip) |
| :heavy_check_mark: | c683e0c3 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6548/4/) | 2025-04-25 10:35:51 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250425-103148-c683e0-pr6548.zip) |
| :heavy_check_mark: | c683e0c3 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6548/4/) | 2025-04-25 10:37:34 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250425-103149-c683e0-pr6548.zip) |
| :heavy_check_mark: | c683e0c3 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6548/4/) | 2025-04-25 10:37:57 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250425-103145-c683e0-pr6548.zip) |
| :heavy_multiplication_x: | c683e0c3 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/4/) | 2025-04-25 10:42:00 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/4//consoleText) |
| :heavy_check_mark: | c683e0c3 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/4/) | 2025-04-25 11:07:56 | ~36 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 35ae4673 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6548/5/) | 2025-05-05 08:01:11 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250505-075830-35ae46-pr6548.zip) |
| :heavy_check_mark: | 35ae4673 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6548/5/) | 2025-05-05 08:01:31 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250505-075827-35ae46-pr6548.aar) |
| :heavy_check_mark: | 35ae4673 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6548/5/) | 2025-05-05 08:03:14 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250505-075839-35ae46-pr6548.zip) |
| :heavy_check_mark: | 35ae4673 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6548/5/) | 2025-05-05 08:03:20 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250505-075832-35ae46-pr6548.zip) |
| :heavy_check_mark: | 35ae4673 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6548/5/) | 2025-05-05 08:04:09 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250505-075835-35ae46-pr6548.zip) |
| :heavy_check_mark: | 35ae4673 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6548/5/) | 2025-05-05 08:04:28 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250505-075831-35ae46-pr6548.zip) |
| :heavy_multiplication_x: | 35ae4673 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/5/) | 2025-05-05 08:09:12 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/5//consoleText) |
| :heavy_check_mark: | 35ae4673 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/5/) | 2025-05-05 08:35:03 | ~36 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 6e9120c8 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6548/6/) | 2025-05-05 09:04:31 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250505-090152-6e9120-pr6548.zip) |
| :heavy_check_mark: | 6e9120c8 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6548/6/) | 2025-05-05 09:04:53 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250505-090151-6e9120-pr6548.aar) |
| :heavy_check_mark: | 6e9120c8 | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6548/6/) | 2025-05-05 09:05:18 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250505-090202-6e9120-pr6548.zip) |
| :heavy_check_mark: | 6e9120c8 | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6548/6/) | 2025-05-05 09:05:56 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250505-090156-6e9120-pr6548.zip) |
| :heavy_check_mark: | 6e9120c8 | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6548/6/) | 2025-05-05 09:07:40 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250505-090158-6e9120-pr6548.zip) |
| :heavy_check_mark: | 6e9120c8 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6548/6/) | 2025-05-05 09:07:45 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250505-090153-6e9120-pr6548.zip) |
| :heavy_multiplication_x: | 6e9120c8 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/6/) | 2025-05-05 09:11:49 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/6//consoleText) |
| :heavy_check_mark: | 6e9120c8 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/6/) | 2025-05-05 09:37:07 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/6//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 5eab2694 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6548/7/) | 2025-05-05 11:35:42 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250505-113240-5eab26-pr6548.zip) |
| :heavy_check_mark: | 5eab2694 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6548/7/) | 2025-05-05 11:35:51 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250505-113231-5eab26-pr6548.aar) |
| :heavy_check_mark: | 5eab2694 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6548/7/) | 2025-05-05 11:36:13 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250505-113236-5eab26-pr6548.zip) |
| :heavy_check_mark: | 5eab2694 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6548/7/) | 2025-05-05 11:37:06 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250505-113232-5eab26-pr6548.zip) |
| :heavy_check_mark: | 5eab2694 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6548/7/) | 2025-05-05 11:38:14 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250505-113236-5eab26-pr6548.zip) |
| :heavy_check_mark: | 5eab2694 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6548/7/) | 2025-05-05 11:39:06 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250505-113233-5eab26-pr6548.zip) |
| :heavy_multiplication_x: | 5eab2694 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/7/) | 2025-05-05 11:42:10 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/7//consoleText) |
| :heavy_check_mark: | 5eab2694 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/7/) | 2025-05-05 12:09:55 | ~37 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 6183e521 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6548/8/) | 2025-05-06 07:38:44 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250506-073601-6183e5-pr6548.zip) |
| :heavy_check_mark: | 6183e521 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6548/8/) | 2025-05-06 07:39:02 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250506-073600-6183e5-pr6548.aar) |
| :heavy_check_mark: | 6183e521 | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6548/8/) | 2025-05-06 07:39:18 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250506-073609-6183e5-pr6548.zip) |
| :heavy_check_mark: | 6183e521 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6548/8/) | 2025-05-06 07:40:03 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250506-073604-6183e5-pr6548.zip) |
| :heavy_check_mark: | 6183e521 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6548/8/) | 2025-05-06 07:41:48 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250506-073606-6183e5-pr6548.zip) |
| :heavy_check_mark: | 6183e521 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6548/8/) | 2025-05-06 07:41:55 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250506-073602-6183e5-pr6548.zip) |
| :heavy_check_mark: | 6183e521 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/8/) | 2025-05-06 07:46:24 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6548/8//consoleText) |
| :heavy_check_mark: | 6183e521 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/8/) | 2025-05-06 08:11:17 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6548/8//consoleText) |

saledjenic committed:
fix(swap)!: remove RouterBuildTransactionsParams and update related functions

The main reason, based on the response we got from the paraswap team, for build tx error that was happening
not always but more or less often, was missed slippage param. In the past slippage parameter was provided at
the later stage, when the tx was about to be built for real, but now, since we fetch the tx data from paraswap api
at the moment we do the gas estimation, the slippage param is necessary earlier. That's why we have to provide
it at the moment of fetching the route and also clients need to re-fetch the route on every slippage change.

What's changed:
- Removed the `RouterBuildTransactionsParams` type and its usage across the codebase.
- Updated `BuildTransactionsFromRoute` function signatures to use `uuid` instead of the removed type.
- Adjusted related database operations and tests to reflect the removal of build input parameters.
- `SwapParaswapProcessor` updated to use slippage param appropriately, reduced calls to paraswap api, by storing tx data

saledjenic commented:
I was checking and additionally fixed:
- when updating the route an empty signal was sent (this was causing an error displayed in the UI for a short moment until the next signal arrives, even though there is no actual error), fixed now
- there was an issue with a negative amountOut value, appropriately fixed now but also added a patch that won't allow negative values for amountOut in the future

Issues like these cannot be solved, cause they are not on our side:
- `{"error":"Internal Error while computing the price"}` you can see it if you open this link https://api.paraswap.io/prices?amount=100000000000&destDecimals=18&destToken=0x6B175474E89094C44Da98b954EedeAC495271d0F&excludeContractMethodsWithoutFeeModel=true&network=1&partner=status.app&side=SELL&srcDecimals=18&srcToken=0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE&userAddress=0x946A89180365d054677D10b1a73235C0Bee6734f&version=6.2
- `context deadline exceeded` may happen, but rarely, due to slow internet speed, high congestion
